sidecar/src/log.rs

```diff
@@ -136,7 +136,32 @@ where
 /// recomputing call site interest all the time.
 /// Ensure that the log level stays the same for at least a few seconds after session disconnect
 /// in order to continue logging the sending of data submitted by the session.
-pub type MultiEnvFilter = TemporarilyRetainedMap<String, EnvFilter>;
+pub struct MultiEnvFilter {
+    map: TemporarilyRetainedMap<String, EnvFilter>,
+    logs_created: Mutex<HashMap<Level, u32>>,
+}
+
+impl MultiEnvFilter {
+    fn default() -> Self {
+        MultiEnvFilter {
+            map: TemporarilyRetainedMap::default(),
+            logs_created: Mutex::new(HashMap::new()),
+        }
+    }
+
+    pub fn add<'a, 's: 'a>(&'s self, key: String) -> TemporarilyRetainedMapGuard<'a, String, EnvFilter> {
+        self.map.add(key)
+    }
+
+    pub fn collect_logs_created_count<'a, 's: 'a>(&'s self) -> HashMap<Level, u32> {
+        let mut map = self.logs_created.lock().unwrap();
+        let clone = map.clone();
+        map.clear();
+
+        clone
```

Comment:
I think `std::mem::take()` would work too, without copying.

Reply:
Looks OK 👌 

But I had to use `std::mem::take(map.deref_mut())` as `map` is a `MutexGuard`. I hope this is just fine

Reply:
Yes, that's correct. I just could not tell you the full codeline off my head without autocompletion :-)